courses_api/api.py

```diff
@@ -18,35 +20,59 @@ class CourseAPIView(viewsets.ReadOnlyModelViewSet):
 
     The API allows for filtering the list of courses.
 
-    * By universities: /api/?university=CNAM&university=CentraleParis
-    * By course subjects: /api/?subject=philosophy&subject=science
-    * By course level: /api/?level=advanced
-    * By availability: 'start-soon', 'end-soon'
-        * /api/?availability=start-soon
+    * By universities: `university=CNAM&university=CentraleParis`
+    * By course subjects: `subject=philosophy&subject=science`
+    * By availability:
+        * `availability=start-soon`
+        * `availability=end-soon`
+        * `availability=enrollment-ends-soon`
+        * `availability=new`
 
     ## Pagination
 
-    You can limit the number of Results Per Page using the rpp API parameter.
+    You can limit the number of Results Per Page using the `rpp`
+    API parameter.
 
-    /api/locations/?rpp=6
+    * Pagination: `/api/courses/locations/?rpp=6`
 
     By default, pagination is set to 10.
 
+    ## Extended List of Courses
+
+    Only courses that are shown in the courses catalog are listed in the
+    public API. Admin API users can access an extended list of courses -
+    courses that are not shown in catalog. When logged in as admin,
+    you can use the `extended_list` parameter.
+
+    * Extended list: `/api/courses/?extended_list=True`
+
     '''
     filter_backends = (CourseFilter,)
     model = Course
     authentication_classes = (TokenAuthentication, SessionAuthentication)
-    serializer_class = CourseSerializer
     paginate_by = 100
     paginate_by_param = 'rpp'
     max_paginate_by = None
 
+    @property
+    def is_admin(self):
+        is_admin = self.request.user.is_staff or self.request.user.is_superuser
```

Comment:
Quand l'utilisateur n'est pas login, il y a quand même un objet `user` attaché à la `request`.
- https://docs.djangoproject.com/en/1.4/topics/auth/#authentication-in-web-requests

Plus explicite :
- https://docs.djangoproject.com/en/1.9/topics/auth/default/#authentication-in-web-requests